<code><h3>Practical Statistics for Data Scientists</h3></code>

<code>Exploratory Data Analysis</code>

In [1]:
import pandas as pd
import numpy as np

In [2]:
data = pd.DataFrame(
    {'State':['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut', 'Delaware'], 
    'Population':[4779736, 710231, 6392017, 2915918, 37253956, 5029196, 3574097, 897934],
    'Murder_rate':[5.7, 5.6, 4.7, 5.6, 4.4, 2.8, 2.4, 5.8],
    'Abbreviation':['AL', 'AK', 'AZ', 'AR', 'CA', 'CO', 'CT', 'DE']
    }
    )
data

State  Population  Murder_rate Abbreviation
0      Alabama     4779736          5.7           AL
1       Alaska      710231          5.6           AK
2      Arizona     6392017          4.7           AZ
3     Arkansas     2915918          5.6           AR
4   California    37253956          4.4           CA
5     Colorado     5029196          2.8           CO
6  Connecticut     3574097          2.4           CT
7     Delaware      897934          5.8           DE

Location Estimates:

<code>Mean, trimmed mean and median</code>

In [3]:
# Mean
data['Population'].mean()

7694135.625

<code>7694135.625</code> is the mean number of books in the <code>Population</code> column

In [4]:
from scipy.stats import trim_mean

In [5]:
# trimmed mean
trim_mean(data['Population'], 0.15)

3931483.0

In [6]:
# median
data['Population'].median()

4176916.5

<code>Weighted Mean</code>

In [7]:
np.average(data['Murder_rate'], weights=data['Population'])

4.376359279149048

<code>Weighted Median</code>

In [15]:
import weighted
weighted.median(data['Murder_rate'], weights=data['Population'])

4.448750497554494

<code>Estimates of Variability</code>

Variance

In [17]:
from statistics import variance

variance(data['Murder_rate'])

1.825

<code>Standard Deviation</code>

In [19]:
data['Murder_rate'].std() # √variance

1.3509256086106296

<code>Percentile</code>

In [21]:
np.percentile(data['Population'], 75)

5369901.25

<code>Mean absolute deviation</code>

In [22]:
from numpy import mean, absolute

mean(absolute(data['Population'] - mean(data['Population'])))

7389955.09375

In [23]:
data.describe()

Population  Murder_rate
count  8.000000e+00     8.000000
mean   7.694136e+06     4.625000
std    1.210575e+07     1.350926
min    7.102310e+05     2.400000
25%    2.411422e+06     4.000000
50%    4.176916e+06     5.150000
75%    5.369901e+06     5.625000
max    3.725396e+07     5.800000

<code>Interquartile Range - IQR</code>

In [24]:
q3, q1 = np.percentile(data['Population'], [75, 25])
iqr = q3 - q1
iqr

2958479.25